# ***# PREDICTING CHANCE OF GETTING ADMISSION***


# ***OBJECTIVE***
**To predicting admission chance of a person.Let considers various factors/variables which influence the chance of getting admission.By using a model to determine chance of person gettion admission**


**DATA SOURCE :https://github.com/YBI-Foundation/Dataset/blob/main/Admission%20Chance.csv**

**IMPORTING LIBRARY**

In [1]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_absolute_percentage_error

**IMPORT DATA AND CREATE DATA FRAME**

In [3]:
df=pd.read_csv('https://github.com/YBI-Foundation/Dataset/raw/main/Admission%20Chance.csv')

**DATA DESCRPTION & VISUALIZATION**

In [4]:
df.head()

Serial No  GRE Score  TOEFL Score  University Rating   SOP  LOR   CGPA  \
0          1        337          118                  4   4.5   4.5  9.65   
1          2        324          107                  4   4.0   4.5  8.87   
2          3        316          104                  3   3.0   3.5  8.00   
3          4        322          110                  3   3.5   2.5  8.67   
4          5        314          103                  2   2.0   3.0  8.21   

   Research  Chance of Admit   
0         1              0.92  
1         1              0.76  
2         1              0.72  
3         1              0.80  
4         0              0.65

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Serial No          400 non-null    int64  
 1   GRE Score          400 non-null    int64  
 2   TOEFL Score        400 non-null    int64  
 3   University Rating  400 non-null    int64  
 4    SOP               400 non-null    float64
 5   LOR                400 non-null    float64
 6   CGPA               400 non-null    float64
 7   Research           400 non-null    int64  
 8   Chance of Admit    400 non-null    float64
dtypes: float64(4), int64(5)
memory usage: 28.2 KB


In [7]:
df.shape

(400, 9)

In [6]:
df.describe()

Serial No   GRE Score  TOEFL Score  University Rating         SOP  \
count  400.000000  400.000000   400.000000         400.000000  400.000000   
mean   200.500000  316.807500   107.410000           3.087500    3.400000   
std    115.614301   11.473646     6.069514           1.143728    1.006869   
min      1.000000  290.000000    92.000000           1.000000    1.000000   
25%    100.750000  308.000000   103.000000           2.000000    2.500000   
50%    200.500000  317.000000   107.000000           3.000000    3.500000   
75%    300.250000  325.000000   112.000000           4.000000    4.000000   
max    400.000000  340.000000   120.000000           5.000000    5.000000   

             LOR         CGPA    Research  Chance of Admit   
count  400.000000  400.000000  400.000000        400.000000  
mean     3.452500    8.598925    0.547500          0.724350  
std      0.898478    0.596317    0.498362          0.142609  
min      1.000000    6.800000    0.000000          0.340000  
25%      3.000000    8.170000    0.000000          0.640000  
50%      3.500000    8.610000    1.000000          0.730000  
75%      4.000000    9.062500    1.000000          0.830000  
max      5.000000    9.920000    1.000000          0.970000

**DATA COLUMNS**

In [8]:
df.columns

Index(['Serial No', 'GRE Score', 'TOEFL Score', 'University Rating', ' SOP',
       'LOR ', 'CGPA', 'Research', 'Chance of Admit '],
      dtype='object')

**TARGET VARIABLE OR DEPENDENT VARIABLE**[link text](https://)

In [11]:
y=df['Chance of Admit ']

In [12]:
y.shape

(400,)

In [13]:
y.head()

0    0.92
1    0.76
2    0.72
3    0.80
4    0.65
Name: Chance of Admit , dtype: float64

**INDEPENDENT VARIABLE OR FEATURES**

In [14]:
X=df[['Serial No', 'GRE Score', 'TOEFL Score', 'University Rating', ' SOP',
       'LOR ', 'CGPA', 'Research']]

In [17]:
X.shape

(400, 8)

In [19]:
X.head()

Serial No  GRE Score  TOEFL Score  University Rating   SOP  LOR   CGPA  \
0          1        337          118                  4   4.5   4.5  9.65   
1          2        324          107                  4   4.0   4.5  8.87   
2          3        316          104                  3   3.0   3.5  8.00   
3          4        322          110                  3   3.5   2.5  8.67   
4          5        314          103                  2   2.0   3.0  8.21   

   Research  
0         1  
1         1  
2         1  
3         1  
4         0

**TRAIN_TEST_SPLIT_DATA**

In [20]:
X_train,X_test,y_train,y_test = train_test_split(X,y,train_size=0.75,random_state=500)

In [22]:
X_train.shape,X_test.shape,y_train.shape,y_test.shape

((300, 8), (100, 8), (300,), (100,))

**MODEL_SELECTION  LINEAR_REGRESSION**

In [34]:
model = LinearRegression()


**BEST FIT LINE**

In [35]:
model.fit(X_train,y_train)

LinearRegression()

In [43]:
model.intercept_

-1.2384606389876824

In [44]:
model.coef_

array([0.00016152, 0.00142021, 0.00420524, 0.00525927, 0.00106163,
       0.02201113, 0.10626717, 0.03159434])

**MODEL_PREDICION**



In [30]:
y_predict = model.predict(X_test)

In [33]:
y_predict

array([0.78644292, 0.9070848 , 0.82615334, 0.93840703, 0.7920011 ,
       0.64710253, 0.64827562, 0.62707135, 0.70296051, 0.64966153,
       0.77886754, 0.79867161, 0.57422191, 0.69560375, 0.66931407,
       0.42012982, 0.97192412, 0.54010676, 0.61599352, 0.84484603,
       0.80011668, 0.79192017, 0.71227003, 0.91846839, 0.5187799 ,
       0.64477731, 0.80137421, 0.80050521, 0.63103995, 0.63703285,
       0.69322354, 0.56439407, 0.56783595, 0.92687226, 0.7952806 ,
       0.86780966, 0.77371954, 0.56144625, 0.87186698, 0.66076128,
       0.72495867, 0.73086811, 0.83863627, 0.48183031, 0.72588524,
       0.81959039, 0.7885037 , 0.74858351, 0.62984682, 0.67593193,
       0.93020212, 0.56219094, 0.7166228 , 0.75995805, 0.63871302,
       0.57403262, 0.77049027, 0.93803728, 0.68613416, 0.66882858,
       0.81955778, 0.6848688 , 0.92515133, 0.79200748, 0.67707761,
       0.86033993, 0.47686816, 0.63644757, 0.83402305, 0.88836326,
       0.71862921, 0.5249864 , 0.83273027, 0.83601854, 0.92357

In [29]:
y_predict.shape

(100,)

**MODEL_EVALUTION**

In [36]:
mean_absolute_percentage_error(y_test,y_predict)

0.06639014091164154

In [37]:
y_test

288    0.82
108    0.93
217    0.85
70     0.94
330    0.80
       ... 
250    0.74
313    0.67
71     0.96
133    0.79
268    0.83
Name: Chance of Admit , Length: 100, dtype: float64

In [38]:
y_predict

array([0.78644292, 0.9070848 , 0.82615334, 0.93840703, 0.7920011 ,
       0.64710253, 0.64827562, 0.62707135, 0.70296051, 0.64966153,
       0.77886754, 0.79867161, 0.57422191, 0.69560375, 0.66931407,
       0.42012982, 0.97192412, 0.54010676, 0.61599352, 0.84484603,
       0.80011668, 0.79192017, 0.71227003, 0.91846839, 0.5187799 ,
       0.64477731, 0.80137421, 0.80050521, 0.63103995, 0.63703285,
       0.69322354, 0.56439407, 0.56783595, 0.92687226, 0.7952806 ,
       0.86780966, 0.77371954, 0.56144625, 0.87186698, 0.66076128,
       0.72495867, 0.73086811, 0.83863627, 0.48183031, 0.72588524,
       0.81959039, 0.7885037 , 0.74858351, 0.62984682, 0.67593193,
       0.93020212, 0.56219094, 0.7166228 , 0.75995805, 0.63871302,
       0.57403262, 0.77049027, 0.93803728, 0.68613416, 0.66882858,
       0.81955778, 0.6848688 , 0.92515133, 0.79200748, 0.67707761,
       0.86033993, 0.47686816, 0.63644757, 0.83402305, 0.88836326,
       0.71862921, 0.5249864 , 0.83273027, 0.83601854, 0.92357

**EXPLANATION :**
THE MODEL PREDICTED CHANCE OF GETTING ADMISSION USING LINEAR REGRESSION MODEL.

1. LIBRARY FILES ARE IMPORTED LIKE PANDAS FOR DATA FRAME,SKLEARN FOR TRAIN_SPLIT_TEST DATA ADN MODELING & EVALUTION.
2. CREATE DATA FRAME TO READ DATA & ANALYZE THE DATA.
3. IDENTIFY THE DEPENDENTED AND INDEPENDENTED VARIABLE FOR FURTHER ANALYSIS
4. SPLIT DATA FOR TRAINING AND TESTING PURPOSE(7:3).
5. SELECT THE MODEL(LINEAR REGRESSION) FOR PREDICTING DATA SINCE TARGET DATA IS CONTINUOUS.
6. DETERMINE THE BEST FIT LINE FOR THE MODEL THEN PREDICT THE VALUE.
7. EVALUATE AND DETERMINE ACCURACY OF THE MODEL.
8. LINEAR REGRESSION USES LINE EQUATION TO PREDICT VALUE
Y = C + MX
Y = TARGET VARIABLE
X = INDEPENDENT VARIABLE
C = INTERCEPT
M = SLOPE
9. NOW C = -1.23846 , SLOPE COEFFICIENT = ([0.00016152, 0.00142021, 0.00420524, 0.00525927, 0.00106163,
       0.02201113, 0.10626717, 0.03159434])


